core/spec/models/spree/promotion_spec.rb

```diff
@@ -64,6 +65,10 @@
     it "should delete rules" do
       expect(Spree::PromotionRule.count).to eq(0)
     end
+
+    it "should delete codes" do
+      expect(Spree::PromotionCode.count).to eq(0)
```

Comment:
yep, i prefer `expect to change` too. to be honest, i wouldn't test `dependent` first of all as @jordan-brough wondered but i'm happy to make the change if we need it.